test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Models/Package/DeprecationCapabilityBaseTests.cs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using NuGet.PackageManagement.UI.Models.Package;
+using NuGet.VisualStudio.Internal.Contracts;
+using Xunit;
+
+namespace NuGet.PackageManagement.UI.Test.Models.Package
+{
+    public class DeprecationCapabilityBaseTests
+    {
+        [Fact]
+        public void IsDeprecated_WithDeprecationMetadata_IsTrue()
+        {
+            // Arrange
+            var deprecationMetadata = new PackageDeprecationMetadataContextInfo("Test message", ["Legacy"], null);
+
+            var capability = new TestDeprecationCapability(deprecationMetadata);
+
+            // Act
+            var result = capability.IsDeprecated;
+
+            // Assert
+            Assert.True(result);
+        }
+
+        [Fact]
+        public void IsDeprecated_WithoutDeprecationMetadata_IsFalse()
+        {
+            // Arrange
+            var capability = new TestDeprecationCapability(null);
+
+            // Act
+            var result = capability.IsDeprecated;
+
+            // Assert
+            Assert.False(result);
+        }
+
+        [Theory]
+        [InlineData(new[] { "CriticalBugs" }, PackageDeprecationReason.CriticalBugs)]
+        [InlineData(new[] { "Legacy" }, PackageDeprecationReason.Legacy)]
+        [InlineData(new[] { "Legacy", "CriticalBugs" }, PackageDeprecationReason.LegacyAndCriticalBugs)]
+        [InlineData(new[] { "Other" }, PackageDeprecationReason.Unknown)]
+        public void PackageDeprecationReasons_MultipleDeprecationReasons_ReturnsExpected(string[] reasons, PackageDeprecationReason expectedMessage)
```

Comment:
It would be good to test case insensitivity since [your code is explicitly case insensitive](https://github.com/NuGet/NuGet.Client/pull/6434/files#diff-a3c7d563e397ad982f4475e11ec83fd4639960b73d30a2c700508aaae33db622R35-R39).